doc/org/build/gameplay.md

```diff
@@ -0,0 +1,722 @@
+
+An entity can only be associated with a single `Object3D`.
+
+You may find it strange that we have a different pattern for `world.eid2obj`, and that we do not simply use the same pattern as the one shown above for `MediaPDF`. Well, I do too. We wrote `world.eid2obj` long before we wrote `MediaPDF`, so this may be an accident. Perhaps we&rsquo;ll change `world.eid2obj` to `Object3DTag.map`, since the `eid2obj` map is meant to be kept in sync with the `Object3DTag` component.
+
+
+<a id="orgfe9fb09"></a>
+
+## Avoid duplicating state
+
+`Object3D` and its subtypes have many properties that change at runtime. Rather than storing a duplicate copy of these properties in `bitECS` components, we use tag components on the entity so that they show up in the necessary queries, and then operate on the associated `Object3D` directly.
+
+For example, `TroikaText` extends `Mesh`, which extends `Object3D`. `TroikaText` s have a `text` string property and a function `sync` that will flush the `text` to the underlying shader program.
+
+In Hubs, we define the `Text` component as a tag (i.e. with no properties):
+
+```typescript
+export const Text = defineComponent();
+```
+
+We do not duplicate the `text` string in a `bitECS` component. We simply operate on the underlying `Object3D` (a `TroikaText`):
+
+```typescript
+const timeLabel = world.eid2obj.get(VideoMenu.timeLabelRef[eid])! as TroikaText;
+timeLabel.text = `${timeFmt(video.currentTime)} / ${timeFmt(video.duration)}`;
+timeLabel.sync();
+```
+
+
+<a id="orgbad6d63"></a>
+
+# Adding entities
+
+
+<a id="org9d285be"></a>
+
+## Entity basics
+
+Entities are added to the world with `addEntity` and removed from the world with `removeEntity`. In `bitECS`, component state is stored in large, pre-allocated `TypedArrays`. In other words, entities are not objects with components inside them. Entities are simply numbers (`EntityID` s), and the `World` keeps track of things like whether an entity has a given component (`hasComponent(world, MyComponent, eid)`).
+
+You will rarely need to call `addEntity` yourself. Instead, you will write entity definitions (`EntityDef` s) using `jsx` or create model files (`gltf`) with that add entities and components when the models are loaded.
+
+Note: We support both `glTF` formats, where binary data buffers contain base64-encoded strings (as in `.gltf`) or raw byte arrays (as in `.glb`). We refer to `gltf` and `glb` files interchangably.
+
+
+<a id="orgdd790f5"></a>
+
+## Creating `EntityDef` s
+
+`EntityDef` s are `jsx` expressions that can be passed to `renderAsEntity` to add entities and components to the world. `EntityDef` s are commonly returned from `template` functions, which take some `InitialData` and return an `EntityDef` with the that `InitialData` applied.
+
+For example, the commonly used `MediaPrefab` function is a `template` that takes `MediaLoaderParams` as its `InitialData`, and returns an `EntityDef` for an interactable object.
+
```

Comment:
I think `MediaPrefab` and `MediaLoader` are bit too complex as a first example. (For me, `MediaLoader` handler is one of the most complex systems in the existing systems.)

I would like to suggest to use a very simple one as a first example to explain `EntityDef`, `Prefab`, `Inflators` here, and use `MediaLoader` in a later section as an advanced one that causes asynchronous work and another entity creation.